### ISSUE TITLE
[Rendering] MS Teams Adaptive Card not rendering full width

### ISSUE THREAD

**Issue by Wajeed-msft** [MicrosoftDocs/msteams-docs#7638]
Copied bug from: https://github.com/microsoft/AdaptiveCards/issues/8102 

**Details from original post:**
### Target Platforms

Other

### SDK Version

1.4

### Application Name

Microsoft Teams

### Problem Description

Adaptive cards do not appear to be rendering full width in Microsoft Teams despite Full Width being declared in the json msteams property.

When the width properly is not declared as full, the card renders at half width. When the width property is declared with the value "full", the card renders at roughly 2/3rd's width. See screenshots for example.

This seems to be consistent across the MS Teams desktop application on Windows 11 and in the MS Teams web application in both Edge and Chrome browsers. This also seems to be a consistent issue regardless of the version declared in the payload json.

In this specific case the card JSON is being POST'ed to a Teams inbound webhook (tested using Thunder Client and also a notification payload from New Relic), I've not been able to test if the same behavior is observed when sending cards from a Logic App.

### Screenshots

![image](https://user-images.githubusercontent.com/40861351/205762049-6bcffc95-b4a1-4956-8483-b8e8bab2d047.png)
This image shows the rendering width of a card without the width full being declared.

![image](https://user-images.githubusercontent.com/40861351/205762201-4633262a-a0a4-4826-a536-be326173a6b2.png)
This image shows the rendering width of the card with width full being declared.

### Card JSON

```JSON
{
  "type": "message",
  "attachments": [
    {
      "contentType": "application/vnd.microsoft.card.adaptive",
      "contentUrl": null,
      "content": {
        "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
        "msteams": {
          "width": "full"
        },
        "$data": {},
        "type": "AdaptiveCard",
        "version": "1.4",
        "body": [
          {
            "type": "TextBlock",
            "text": "For Samples and Templates, see [https://adaptivecards.io/samples](https://adaptivecards.io/samples)"
          },
          {
            "type": "Image",
            "url": "https://adaptivecards.io/content/cats/1.png"
          }
        ]
      }
    }
  ]
}
```

### Sample Code

_No response_---
#### Document Details

⚠ *Do not edit this section.It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 3fb4394d-25b1-8752-3996-fe3a2934732d
* Version Independent ID: 6d391381-89d0-117c-27cc-00043bd26a36
* Content: [Cards reference - Teams](https://docs.microsoft.com/en-us/microsoftteams/platform/task-modules-and-cards/cards/cards-reference)
* Content Source: [msteams-platform/task-modules-and-cards/cards/cards-reference.md](https://github.com/MicrosoftDocs/msteams-docs/blob/master/msteams-platform/task-modules-and-cards/cards/cards-reference.md)
* Product: **msteams**
* GitHub Login: @surbhigupta12
* Microsoft Alias: **surbhigupta12**

**Comment by mikesprague** [comment]
Adding comment from https://github.com/microsoft/AdaptiveCards/issues/8102#issuecomment-1340965031:

I'm seeing this behavior as well. Previously full-width AdaptiveCards are now rendering at a fixed width.

Using dev tools to inspect one of my AdaptiveCards in Teams, I see the following:

```css
.ts-adaptive-card.adaptive-card-responsive.adaptive-card-fullwidth {
    width: 60.6rem;
    max-width: 100%;
}
```

When I remove (or uncheck) the width attribute in that style the card shows full-width again. Not sure if that style comes from Teams or some AdaptiveCard tooling but that is causing the issue. 

This was working as expected until 24-48 hours ago.

**Comment by PavanMudigondaTR** [comment]
Adaptive Card is not rendering full sized from yesterday after 6 pm EST. All previously sent cards with same script that appeared full size started to appear short now. I thought it would start working morning but its not working. I just found this thread. Mine Is MacOS and I reinstalled MS Teams but that didn’t resolve the problem.

**Comment by Prasad-MSFT** [comment]
@mikesprague , @PavanMudigondaTR - We are able to repro this issue. We have raised a bug for the same and engineering team is looking into it. We will inform you once we get any further update. Thanks!
**Card without width full being declared:**
![image](https://user-images.githubusercontent.com/62237649/206369634-e0841b55-34f6-41fd-a4c2-082ee668d049.png)

**Card with width full being declared:**
![image](https://user-images.githubusercontent.com/62237649/206369708-b4b8d3a3-8174-468b-95d7-036ab58e8154.png)

**Comment by PavanMudigondaTR** [comment]
Please provide an update or an eta, if its possible !

**Comment by Prasad-MSFT** [comment]
@PavanMudigondaTR - Apologies. We are checking the status of the bug with engineering team. Currently there is no ETA to share. We will inform you once we get any update from engineering team. Thanks!

**Comment by WilbertWalrus** [comment]
> @PavanMudigondaTR - Apologies. We are checking the status of the bug with engineering team. Currently there is no ETA to share. We will inform you once we get any update from engineering team. Thanks!

Understand the engineering team is off on holidays and this is a low priority issue, but can you guys at Microsoft do proper back-testing when rolling out new changes to the Adaptive Card so that basic layouts such as "full width" does not get messed up? Thanks!

**Comment by PavanMudigondaTR** [comment]
> > @PavanMudigondaTR - Apologies. We are checking the status of the bug with engineering team. Currently there is no ETA to share. We will inform you once we get any update from engineering team. Thanks!
> 
> 
> 
> Understand the engineering team is off on holidays and this is a low priority issue, but can you guys at Microsoft do proper back-testing when rolling out new changes to the Adaptive Card so that basic layouts such as "full width" does not get messed up? Thanks!

I concur. I suggest to use Microsoft Playwright for doing simple Visual Test. If Microsoft breaks teams adaptive card again next time you will be troubling everyone again and we will have to keep waiting for months to get fix.

**Comment by PavanMudigondaTR** [comment]
@Prasad-MSFT 

Could you please provide an update or ETA now that holidays are over !

**Comment by srikanth2042** [comment]
I filed an issue about exactly the same problem https://github.com/MicrosoftDocs/msteams-docs/issues/7468. Waiting for a fix from the Microsoft team.

**Comment by Prasad-MSFT** [comment]
Hi @PavanMudigondaTR , @srikanth2042 - We are proactively following up with engineering team about the update. 
We will let you know as soon as we get any input from engineering team.
Thanks!

**Comment by xiphoid24** [comment]
@Prasad-MSFT when it comes to determining priority, I think it is worth noting this is not a feature request. This issue is a bug that was introduced with an update to MS Teams. Previously, adaptive cards would render in actual full width.

**Comment by mikesprague** [comment]
@Prasad-MSFT I would also like to reiterate that this is a bug for a documented feature (https://learn.microsoft.com/en-us/microsoftteams/platform/task-modules-and-cards/cards/cards-format?tabs=adaptive-md%2Cdesktop%2Cconnector-html#construct-full-width-cards).

A (previously working) documented feature that has now been broken for over a month.

The limited updates and clear lack of prioritization is disappointing to say the least 😞

**Comment by PavanMudigondaTR** [comment]
> @Prasad-MSFT I would also like to reiterate that this is a bug for a documented feature (https://learn.microsoft.com/en-us/microsoftteams/platform/task-modules-and-cards/cards/cards-format?tabs=adaptive-md%2Cdesktop%2Cconnector-html#construct-full-width-cards).
> 
> A (previously working) documented feature that has now been broken for over a month.
> 
> The limited updates and clear lack of prioritization is disappointing to say the least 😞

To add on to this, this appears to be repeating bug as well as I saw similar bugs reported earlier as well.

**Comment by PavanMudigondaTR** [comment]
How come this is not been triaged yet. Even after a month ! How long will it just sit there in pending triage ?

**Comment by Wajeed-msft** [comment]
@PavanMudigondaTR - Apologies for delays. We have shared repro steps and all the required details with engineering team to investigate and fix the bug. We will keep you updates as soon as we get further information.

**Comment by Wajeed-msft** [comment]
**Update:**  Team has identified the issue and the fix is being reviewed internally. We will keep you posted with updates on the status.

**Comment by Wajeed-msft** [comment]
**Update:** The fix would start rolling out next week and expected to hit public client builds in 3 to 4 weeks post that. 
Note that this is tentative ETA and could change. We will keep you posted with updates.

**Comment by Prasad-MSFT** [comment]
**Update:**
@PavanMudigondaTR , @mikesprague , @xiphoid24 - The fix is rolling out in internal rings and will roll out to public ring in 1-2 weeks. Currently there is no exact ETA to share. Thanks!

**Comment by sinansprinklr** [comment]
Hi @Prasad-MSFT Could you please help us with an ETA for the above fix? This has been persisting for quite a while now.

**Comment by SinanRahoof** [comment]
Hi @Prasad-MSFT could you please let us know if there is any update on the above issue;? Or any idea about the Estimated date when this patch goes live??

**Comment by Viajaz** [comment]
What will be done in the future to prevent a recurrence of bugs like this? We've had to rewrite software and workflows because this issue has gone on for months.

**Comment by Prasad-MSFT** [comment]
@SinanRahoof, @Viajaz - Apologies. The fix is rolling out in internal rings and will roll out to public ring by end of February.
We will update this thread by testing once it is released publicly.

**Comment by barth11** [comment]
Will this work both in Compact and Comfy view?

**Comment by SinanRahoof** [comment]
@Prasad-MSFT will this new change go live today? As you mentioned above? It's March 1 today. My clients have been asking me for updates on this!

**Comment by PavanMudigondaTR** [comment]
@Prasad-MSFT I am not seeing the fix working either today. Please help !

**Comment by PavanMudigondaTR** [comment]
@Prasad-MSFT  please provide an update or a tentative release date

**Comment by timeverts** [comment]
100% agree. This is a pretty critical bug fix.

@Prasad-MSFT, please advise of the public release date?

**Comment by SinanRahoof** [comment]
Hi @Prasad-MSFT, could you please let us know when the fix would go live. Another week passed by. I totally respect that the team is working hard on the fix. However, since the previous update, we had made commitments to our client on the fix for the change in width.

I totally understand your team would now be over burdened with the fix pressure. Could you please give me a final tentative so that I could give my team and clients an expectation when this would go live?

**Comment by mikesprague** [comment]
@Prasad-MSFT @Wajeed-msft @ChetanSharma-msft 

I've not piled on yet because I haven't expected any answers based on how this issue has gone so far...

This is all extremely disappointing. Many of us have chosen to embrace your technologies and "standards" (such as adaptive cards), and then taken the time to develop interactive customer facing notifications that rely on certain properties to display effectively and provide the customer a good UI/UX, only to have you (the vendor) break the spec and put NO priority on fixing it for those customers using it with MS Teams (another vendor technology/platform). 

You have my team fully reconsidering our choices and debating rebuilding using other tools altogether.

Looking back at this issue, it sat for weeks (months, really) and then all of the sudden it was "fixed" and would roll out in a couple weeks. Those updates never felt genuine at the time given the timing and previous lack of updates.

I would recommend commenting on this soon and perhaps either retiring the Adaptive Card spec or making sure you are fully supporting your own stuff.

**Comment by srikanth2042** [comment]
I am not sure if Microsoft is doing proper testing related to Adaptive Cards before releasing any new updates. With the introduction of Channels 2.0, more bugs are coming our way https://github.com/MicrosoftDocs/msteams-docs/issues/8091. If our clients start adapting to this new feature, we are going to have many more complaints from them.
![image](https://user-images.githubusercontent.com/33093586/223775737-29f8ed35-e744-4992-a341-c4a677c303cf.png)

This is really frustrating, we really need to reconsider using Adaptive Cards.

**Comment by PavanMudigondaTR** [comment]
@Prasad-MSFT @Wajeed-msft @ChetanSharma-msft

Please provide some update or a tentative date for fix.

**Comment by SinanRahoof** [comment]
@Prasad-MSFT @Wajeed-msft @ChetanSharma-msft please don't rat us out. If it's taking too much time, please be transparent with us. We have commitments with our clients and we don't want to disappoint them.

**Comment by Wajeed-msft** [comment]
@SinanRahoof, @PavanMudigondaTR - I have escalated this to engineering team. Checking current status on this.

**Comment by Wajeed-msft** [comment]
@SinanRahoof, @PavanMudigondaTR  - apologies for the confusion. Fix for this was not merged earlier. Engineering team is going to merge the fix today. I'll update the thread once it's done. Based on the merge tentative ETA would be around end of April. We will share the updates once fix started rolling out to keep everyone updated. Thank you for your patience.

**Comment by PavanMudigondaTR** [comment]
> @SinanRahoof, @PavanMudigondaTR - apologies for the confusion. Fix for this was not merged earlier. Engineering team is going to merge the fix today. I'll update the thread once it's done. Based on the merge tentative ETA would be around end of April. We will share the updates once fix started rolling out to keep everyone updated. Thank you for your patience.

@Wajeed-msft 
This is unacceptable...how come you are taking 5 months for patching an already documented functionality that you broken with suddenly. This is not a new feature any how and it renders the whole Adaptive card unusable as well. My team cursing me for using Adaptive Card Solution. The Message Card is a stable functionality than this.

**Comment by alhnemer** [comment]
@Prasad-MSFT @Wajeed-msft @ChetanSharma-msft

Please provide some update or a tentative ETA. 
This issue prevents us from using the "Adaptive Card" solution.

_FYI :_ @FlorentLAVAUD-Sage @pierrickpailleaud @pirem

**Comment by PavanMudigondaTR** [comment]
> @Prasad-MSFT @Wajeed-msft @ChetanSharma-msft
> 
> Please provide some update or a tentative ETA. This issue prevents us from using the "Adaptive Card" solution.
> 
> _FYI :_ @FlorentLAVAUD-Sage @pierrickpailleaud @pirem @anemer

they forgot to merge the fix in early Jan so the release date of Feb end now became end of April. This Is what I understood. We can expect this fix coming in month of May and potentially get delayed more as there is chance they can forget again.

**Comment by Wajeed-msft** [comment]
**Update:** Fix for this issue is merged. I am checking with engineering team to expedite the fix rollout.

**Comment by PavanMudigondaTR** [comment]
> **Update:** Fix for this issue is merged. I am checking with engineering team to expedite the fix rollout.

So it’s been a week, could you please provide an update on this if you got response from your Engineering Team ? @Wajeed-msft

**Comment by alhnemer** [comment]
> @Prasad-MSFT @Wajeed-msft @ChetanSharma-msft
> 
> Please provide some update or a tentative ETA. This issue prevents us from using the "Adaptive Card" solution.
> 
> _FYI :_ @FlorentLAVAUD-Sage @pierrickpailleaud @pirem @anemer

Hello,

any update?

**Comment by Wajeed-msft** [comment]
@PavanMudigondaTR, @alhnemer, @SinanRahoof, @mikesprague, @srikanth2042 - could you please share the Ids of tenants which are most impacted so that we could roll out an early fix to these tenants?

**Comment by PavanMudigondaTR** [comment]
@Wajeed-msft I do not have Tenant ID and also I can't share it here. Our org is Thomson Reuters and you will find our account easily.

**Comment by Wajeed-msft** [comment]
@mikesprague , @PavanMudigondaTR  - Please share it on [Microsoft Teams Community Help](mailto:microsoftteamsdev@microsoft.com) alias.

**Comment by Viajaz** [comment]
I have emailed you
@Wajeed-msft

**Comment by PavanMudigondaTR** [comment]
> @mikesprague , @PavanMudigondaTR - You can share it on [Microsoft Teams Community Help](mailto:microsoftteamsdev@microsoft.com) alias.

@Wajeed-msft  I was able to find the Tenant ID and I have sent you email to your [Microsoft Teams Community Help](mailto:microsoftteamsdev@microsoft.com)

**Comment by Prasad-MSFT** [comment]
@Viajaz , @PavanMudigondaTR  - Thanks for sharing tenant details. We have passed on these to engineering team. We will inform you once fix is done.

**Comment by SinanRahoof** [comment]
Hi @Wajeed-msft , i saw this right now. And i have forwarded the tenant id on the email that you have given above. 

Please free to let me know in case if you require any additional information from my end on this.

**Comment by Prasad-MSFT** [comment]
@SinanRahoof - Thanks for sharing the details. We will pass this to engineering team and will update you once fix is done.

**Comment by soseki-angfa** [comment]
@Wajeed-msft 
Thank you very much for fixing this issue.
I have a question regarding this.
Do we have to change the schema verion if we want to apply this fix?
I am talking about following.
```
            "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
            "version": "1.5",
```

Regards,

**Comment by PavanMudigondaTR** [comment]
We are at end of April. Are we getting the fix ? @Prasad-MSFT

**Comment by Prasad-MSFT** [comment]
Apologies! 
Currently the fix is published and available in internal ring.  Once its publicly released we will update the thread.

**Comment by Prasad-MSFT** [comment]
> @Wajeed-msft Thank you very much for fixing this issue. I have a question regarding this. Do we have to change the schema verion if we want to apply this fix? I am talking about following.
> 
> ```
>             "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
>             "version": "1.5",
> ```
> 
> Regards,

Hi @soseki-angfa - It doesn't have a 1.5 version dependency.

**Comment by soseki-angfa** [comment]
@Prasad-MSFT 
Okay :)
Thanks for reply.

**Comment by Prasad-MSFT** [comment]
Hello Everyone, the issue is fixed now. Tested in MS Teams Desktop (Microsoft Teams Version 1.6.00.13566) and Web version.
Please do verify.

![image](https://github.com/MicrosoftDocs/msteams-docs/assets/62237649/f51da273-d34c-487f-9247-d1bd5035d58b)

**Comment by jaredbrogan** [comment]
> Hello Everyone, the issue is fixed now. Tested in MS Teams Desktop (Microsoft Teams Version 1.6.00.13566) and Web version. Please do verify.

Looks good on my end as well. Thanks for following through on this!

**Comment by Prasad-MSFT** [comment]
Thanks everyone for confirming. We are closing this issue for now. Please feel free to reopen if required. 

Thanks again!

**Comment by Prasad-MSFT** [comment]
Could you please share your valuable feedback via [Microsoft Teams Developer Community Response Feedback ](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR7iCOpS5_b9Nqmwx43u5rtZUOThVR081SllSR05aSDQxQ0tUMDVPTVIxTi4u)link?

**Comment by AhmedWaelH** [comment]
Hello @prasad-MSFT @Wajeed-msft ,
I tried to get the version mentioned "Version 1.6.00.13566" for windows desktop app, but found that the last available version is 1.6.00.11166, can you please help in getting this version for teams "Version 1.6.00.13566"

**Comment by Prasad-MSFT** [comment]
@AhmedWaelH - This is the link to downgrade or upgrade Microsoft Teams
https://microsoft-teams.en.uptodown.com/windows/versions

**Comment by AhmedWaelH** [comment]
Thanks @Prasad-MSFT for your response,

but know I am a little bit confused, because from this site [https://learn.microsoft.com/en-us/officeupdates/teams-app-versioning](url) it mentions that the latest version is 1.6.00.11166, and the link you provided above mentions that the last available version is 1.6.00.1381, so can you please clarify for me the gap in the understanding, that I got, 

also does this version 1.6.00.11166 got the solution for the thread ticket, or only this version 1.6.00.1381 as it is newer than 1.6.00.13566

**Comment by Prasad-MSFT** [comment]
@AhmedWaelH - The fix has been publicly rolled out. We have tested in `Microsoft Teams Version 1.6.00.13566`. You can verify in `Teams Version 1.6.00.11166`. If that doesn't work you can verify in other mentioned version.
